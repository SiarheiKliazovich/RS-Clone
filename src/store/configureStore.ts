/* eslint-disable @typescript-eslint/ban-ts-comment */
import { applyMiddleware, createStore } from "redux";
import thunkMiddleware from "redux-thunk";

import loggerMiddleware from "./logger";
import rootReducer from "../reducers";

const configureStore = () => {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(rootReducer, middlewareEnhancer);

  // @ts-ignore:next-line
  if (process.env.NODE_ENV !== "production" && module.hot) {
    // @ts-ignore:next-line
    module.hot.accept("./../reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
};

export default configureStore;
