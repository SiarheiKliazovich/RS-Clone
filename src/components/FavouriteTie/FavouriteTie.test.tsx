import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import FavoriteTie from ".";

const mockStore = configureStore([]);

describe("FavoriteTie", () => {
  // eslint-disable-next-line
  let store: any;

  beforeEach(() => {
    store = mockStore({
      appInterface: {
        accentColor: {
          static: "#000",
          hover: "#fff",
        },
      },
      langReducer: {
        lang: "en",
      },
      favouriteTieReducer: {
        addTieLoadingStatus: "load",
      },
      auth: {
        user: {
          role: "USER",
        },
      },
    });
  });

  test("renders correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <FavoriteTie />
        </Router>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
