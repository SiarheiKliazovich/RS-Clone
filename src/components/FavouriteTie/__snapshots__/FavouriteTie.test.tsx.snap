// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteTie renders correctly 1`] = `
<div>
  <main
    class="main-market"
  >
    <div
      class="tiemarket-wrapper"
    >
      <div
        class="container"
      >
        <div
          class="tiemarket__banner-wrapper"
        >
          <div
            class="market-block"
          >
            <div
              class="market-block__title_wrapper"
            >
              <h4
                class="market-block__title"
              >
                My wishlist
              </h4>
            </div>
            <div
              class="market-block__products"
            >
              <div
                class="row__products"
              >
                <p>
                  You didn't add ties in your wishlist
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="popup-overlay"
    >
      <div
        class="popup"
      >
        <div
          class="sign__content"
        >
          <h2
            class="sign__title"
          >
            sign.signIn
          </h2>
          <div
            class="sign__text"
          >
            sign.text-1
          </div>
          <form
            action="#"
            class="form"
          >
            <input
              class="field"
              id="email"
              name="email"
              placeholder="Email"
              type="email"
              value=""
            />
            <input
              class="field"
              id="password"
              name="password"
              placeholder="Password"
              type="password"
              value=""
            />
            <button
              class="sign__button"
              style="background-color: rgb(0, 0, 0);"
              type="submit"
            >
              sign.signIn
            </button>
          </form>
          <div
            class="sign__text-link"
          >
            sign.create
             
            <a
              class="text-link"
              href="/sign-up"
              style="color: rgb(0, 0, 0);"
            >
              sign.signUp
            </a>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
